File: app/src/main/java/com/attendance/data/source/local/CoursesLocalDataSource.java
package com.attendance.data.source.local;

import android.content.Context;

import com.attendance.data.Course;
import com.attendance.data.CourseDao;
import com.attendance.data.DaoMaster;
import com.attendance.data.DaoSession;

import org.greenrobot.greendao.query.QueryBuilder;

import java.util.List;

/**
 * Created by peiqin on 9/18/2016.
 */
public class CoursesLocalDataSource {

    private CourseDao courseDao;

    public CoursesLocalDataSource(Context context) {
        DBHelper dbHelper = DBHelper.getInstance(context);
        DaoMaster daoMaster = new DaoMaster(dbHelper.getWritableDatabase());
        DaoSession daoSession = daoMaster.newSession();
        courseDao = daoSession.getCourseDao();
    }

    public List<Course> findAll() {
        QueryBuilder<Course> qb = courseDao.queryBuilder();
        List<Course> list = qb.list();
        return list;
    }

    public void insert(Course course) {
        courseDao.insert(course);
    }

    public void delAll() {
        courseDao.deleteAll();
    }

}

File: app/src/main/kotlin/com.attendance/data/source/local/CoursesLocalDataSource.kt
package com.attendance.data.source.local

import android.content.Context
import android.database.Cursor
import android.database.sqlite.SQLiteDatabase
import com.attendance.data.Course
import com.attendance.utils.DBHelper

/**
 * Created by peiqin on 9/18/2016.
 */
class CoursesLocalDataSource(context: Context) {

    private val db: SQLiteDatabase

    init {
        db = DBHelper.getInstance(context)
    }

    fun findAll(): Cursor {
        val cursor = db.rawQuery("select * from course order by id desc", null)
        return cursor
    }

    fun insert(id: Int, name: String) {
        db.execSQL("insert into course values('$id','$name')")
    }

    fun delAll() {
        db.execSQL("delete from course")
    }

}